### PR TITLE
Update @gear-api examples

### PR DESCRIPTION
**Pull request by EugenWay**
https://github.com/gear-tech/wiki/pull/112

### CODE REVIEW

File: docs/developing-contracts/gear-program.md [hunk 1]
```diff
@@ -5,25 +5,263 @@ sidebar_position: 2
...
-    msg::reply(b"Hello world!", 0, 0);
+    let action: Action = msg::load().unwrap();
+
+    debug!("Received action: {:?}", action);
+
+    match action {
+        Action::AddCandidate(name) => {
+            STATE.add_candidate(name.clone());
+
+            msg::reply((), 0, 0);
```
Comment (shamilsan) [inline docs/developing-contracts/gear-program.md:178]: New `gstd` hasn't gas limit in `reply()` function.
```suggestion
            msg::reply((), 0);
```

File: docs/developing-contracts/gear-program.md [hunk 2]
```diff
@@ -5,25 +5,263 @@ sidebar_position: 2
...
+
+            debug!("Added new candidate: {:?}", name);
+        }
+
+        Action::VoteForCandidate(name) => {
+            STATE.vote_for_candidate(name.clone());
+
+            msg::reply((), 0, 0);
```
Comment (shamilsan) [inline docs/developing-contracts/gear-program.md:186]:
```suggestion
            msg::reply((), 0);
```

File: docs/developing-contracts/gear-program.md [hunk 3]
```diff
@@ -5,25 +5,263 @@ sidebar_position: 2
...
+               .expect("Can't find any candidate");
+ 
+           StateReply::VotesFor(votes_for_candidate.clone()).encode()
+       }
+   };
+ 
+   let result = gstd::macros::util::to_wasm_ptr(&encoded[..]);
+   core::mem::forget(encoded);
+ 
+   result
```
Comment (shamilsan) [inline docs/developing-contracts/gear-program.md:218]: It is recommended to use a new function for getting a pointer.
https://github.com/gear-tech/gear/blob/master/examples/meta/src/lib.rs#L167
```suggestion
   gstd::util::to_leak_ptr(encoded)
```

File: docs/developing-contracts/gear-program.md [hunk 4]
```diff
@@ -5,25 +5,263 @@ sidebar_position: 2
...
+cargo new voting-app --lib
+```
+
+Take look at project structure:
+
+```
+voting-app/
+├── Cargo.toml
+└── src
+_____└── lib.rs
```
Comment (shamilsan) [inline docs/developing-contracts/gear-program.md:54]:
```suggestion
     └── lib.rs
```

File: docs/developing-contracts/gear-program.md (hunk 1 above)
Comment (EugenWay) [reply docs/developing-contracts/gear-program.md]: This tutorial was pushed by mistake

File: docs/developing-contracts/gear-program.md (hunk 2 above)
Comment (EugenWay) [reply docs/developing-contracts/gear-program.md]: This tutorial was pushed by mistake

**Review by shamilsan — APPROVED** [review]
The Rust part is ok to me.

**Comment by xrdavies** [comment]
> The Rust part is ok to me.

@shamilsan  Hi Shamil, seems you need to resolve those conversions.

**Comment by shamilsan** [comment]
> @shamilsan Hi Shamil, seems you need to resolve those conversions.

Done.

**Pull request merged**